compilers/concrete-compiler/compiler/lib/Runtime/wrappers.cpp

```diff
@@ -1011,3 +1012,20 @@ void memref_trace_message(char *message_ptr, uint32_t message_len) {
   std::string message{message_ptr, (size_t)message_len};
   std::cout << message << std::flush;
 }
+
+void *concrete_checked_malloc(size_t size) {
```

Comment:
maybe you could having a test, where you would concrete_checked_malloc a gigantic size, and check this is caught well? and the same thing at the CP level, to check our users see they ask too much memory, since we saw that kind of bugs recently

Reply:
I added a unit test for this (I didn't knew we can test failures in gtests before). However, it's not obvious for CP as it really depends on the machine being tested. For example, the bug we were debugging initially was causing the process to be killed, while on other systems, it was a bad alloc. I tried making a test locally, but I can't get passed the process being killed, and even if it works locally, I'm not sure it's gonna work in the CI.
Anyway, I think this unittest is enough, since it's testing the allocation function used in the generated code.

Reply:
ok yes, I understand